File: common/src/main/java/generations/gg/generations/core/generationscore/network/packets/dialogue/C2SCloseDialoguePacket.java
package generations.gg.generations.core.generationscore.network.packets.dialogue;

import generations.gg.generations.core.generationscore.network.packets.GenerationsNetworkPacket;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.resources.ResourceLocation;
import org.jetbrains.annotations.NotNull;

import static generations.gg.generations.core.generationscore.GenerationsCore.id;

public class C2SCloseDialoguePacket implements GenerationsNetworkPacket<C2SCloseDialoguePacket> {
    @Override
    public ResourceLocation getId() {
        return ID;
    }

    @Override
    public void encode(@NotNull FriendlyByteBuf friendlyByteBuf) {

    }

    public static ResourceLocation ID = id("close_dialgoue");
    public static C2SCloseDialoguePacket decode(FriendlyByteBuf buf) {
        return new C2SCloseDialoguePacket();
    }

}

File: common/src/main/java/generations/gg/generations/core/generationscore/network/packets/dialogue/C2SCloseDialoguePacket.kt
package generations.gg.generations.core.generationscore.network.packets.dialogue

import com.cobblemon.mod.common.api.net.NetworkPacket
import com.cobblemon.mod.common.api.net.ServerNetworkPacketHandler
import generations.gg.generations.core.generationscore.GenerationsCore.id
import generations.gg.generations.core.generationscore.network.packets.C2SToggleCookingPotPacket
import generations.gg.generations.core.generationscore.world.dialogue.DialogueManager
import net.minecraft.network.FriendlyByteBuf
import net.minecraft.server.MinecraftServer
import net.minecraft.server.level.ServerPlayer

class C2SCloseDialoguePacket : NetworkPacket<C2SCloseDialoguePacket> {
    override val id = ID

    override fun encode(buf: FriendlyByteBuf) {}

    companion object {
        val ID = id("close_dialgoue")
        fun decode(buf: FriendlyByteBuf) = C2SCloseDialoguePacket()
    }

    class Handler: ServerNetworkPacketHandler<C2SCloseDialoguePacket> {
        override fun handle(packet: C2SCloseDialoguePacket, server: MinecraftServer, player: ServerPlayer) {
            if (DialogueManager.DIALOGUE_MAP.containsKey(player)) DialogueManager.DIALOGUE_MAP[player]!!.discard()
        }

    }
}
